### ISSUE TITLE
Does not accept URLs with underscores

### ISSUE THREAD

**Issue by kmlucy** [linuxserver/Heimdall#116]
If I try to add a URL with an underscore, it kicks back an invalid url error.

**Comment by kmlucy** [comment]
It's for my printer: http://brother_laser.subdomain.domain.com. If I take out the underscore, Heimdall accepts it.

**Comment by KodeStar** [comment]
interesting

**Comment by KodeStar** [comment]
https://bugs.php.net/bug.php?id=64948 looks like it's a bug with the validator

**Comment by norweeg** [comment]
technically it is recommended you use hyphens instead of undercores.  If you google "underscore in url", you will get a ton of pages recommend you not use them

https://www.seomechanic.com/seo-101-hyphens-underscores-_-urls/

**Comment by mueslo** [comment]
> I'm going to close this as underscores aren't technically valid so the validator is correct in rejecting it. The alternative is to use the IP I guess, or a tinyurl that redirects.

This is wrong. They are perfectly valid domain names: https://stackoverflow.com/questions/2180465/can-domain-name-subdomains-have-an-underscore-in-it

**Comment by norweeg** [comment]
@mueslo while the standard allows for them, it also recommends against non-alphanumeric characters in domains because they are not considered safe.  See RFC 1178 https://tools.ietf.org/html/rfc1178
